src/templates/CustomTypeTemplate.java

```diff
@@ -0,0 +1,131 @@
+{%- let package_name = config.package_name() %}
+{%- match config.custom_types.get(name.as_str())  %}
+{%- when None %}
+{#- Define the type using typealiases to the builtin #}
+{%- let ffi_type_name=builtin|ffi_type|ffi_type_name_by_value %}
+
+ package {{ package_name }};
+
+// Newtype
+public record {{ type_name }}(
+  {{ builtin|type_name(ci) }} value
+) {
+}
+
+package {{ package_name }};
+import java.nio.ByteBuffer;
+
+public enum {{ ffi_converter_name }} implements FfiConverter<{{ type_name }}, {{ ffi_type_name}}> {
+  INSTANCE;
+  @Override
+  public {{ type_name }} lift({{ ffi_type_name }} value) {
+      var builtinValue = {{ builtin|lift_fn }}(value);
+      return new {{ type_name }}(builtinValue);
+  }
+  @Override
+  public {{ ffi_type_name }} lower({{ type_name }} value) {
+      var builtinValue = value.value();
+      return {{ builtin|lower_fn }}(builtinValue);
+  }
+  @Override
+  public {{ type_name }} read(ByteBuffer buf) {
+      var builtinValue = {{ builtin|read_fn }}(buf);
+      return new {{ type_name }}(builtinValue);
+  }
+  @Override
+  public long allocationSize({{ type_name }} value) {
+      var builtinValue = value.value();
+      return {{ builtin|allocation_size_fn }}(builtinValue);
+  }
+  @Override
+  public void write({{ type_name }} value, ByteBuffer buf) {
+      var builtinValue = value.value();
+      {{ builtin|write_fn }}(builtinValue, buf);
+  }
+}
+
+{%- when Some with (config) %}
+
+{%- let ffi_type_name=builtin|ffi_type|ffi_type_name_by_value %}
```

Comment:
This is the same in both branches of this match right? I think it can get pulled up next to the `package_name` one.